src/Avalonia.Controls/TopLevel.cs

```diff
@@ -507,12 +506,20 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
         /// <param name="e">The event args.</param>
         private void HandleInput(RawInputEventArgs e)
         {
+            if (e.Device != _lastActivePointerDevice)
+            {
+                ClearPointerOver();
+            }
+
+            // Set last active device before processing input, because ClearPointerOver might be called and clear last device.
+            _lastActivePointerDevice = e.Device as IPointerDevice;
```

Comment:
I was thinking to add pointer device argument to the SetPointerOver method, so it can be saved in top-level and used in SceneInvalidated. But decided to keep it internal here. Couldn't find any other reasons for the public argument.